src/components/AutoComplete/AutoComplete.js

```diff
@@ -95,7 +95,9 @@ export default class AutoComplete extends Component {
 
   render() {
     const { value } = this.state;
-    const { id, source } = this.props;
+    const { source } = this.props;
+    let { id } = this.props;
+    id = `${id}-${Math.random()}`;
```

Comment:
Sorry. Is this relevant to `prevent autocomplete`?
Also, why indeed you need to generate random id (and if so, why the component is doing this instead of the component which using the `AutoComplete`?

And last note, if you're override the `id` variable, why do you need to declare it (using destruction) from the first place?

Instead of
```
let { id } = this.props;
id = `${id}-${Math.random()}`;
```

Just

```
id = `${id}-${Math.random()}`;
```

Reply:
Thank you for the clarification 😄 

> Sorry. Is this relevant to prevent autocomplete?

It is, yes.  Regrettably, there's a [long and storied](https://stackoverflow.com/questions/12374442/chrome-ignores-autocomplete-off) past on disabling `autocomplete` across all browsers; Chrome is particularly aggressive on this.

---

> Also, why indeed you need to generate random id (and if so, why the component is doing this instead of the component which using the AutoComplete?

Over the years I've tried nearly all of the approaches suggested in the SO posting I linked, as well as things suggested in the adjacent articles.  The tactic that seems to be the most reliable and that creates the smallest surface area for issues is to randomize the `name` attribute which, for completeness, is taken from the `id` if `name` is not provided.

This is done in the `<AutoComplete>` component since the goal of the component is to provide a controlled, browser-agnostic autocomplete experience.  As such, it felt to me like the correct point of abstraction for this functionality.  Put another way, I wanted to create a guarantee for any engineer that uses this `<AutoComplete>` component.  That guarantee can be stated as:

*"If you use the `<AutoComplete>` component, you can be assured that the **browser's** `autocomplete` functionality will be disabled on the underlying input."*

---

> And last note, if you're override the id variable, why do you need to declare it (using destruction) from the first place?

My intent is not to override the the `id` prop; it is to extend the `id` prop with a randomized string.  To accomplish this, I need access to the incoming `id` prop so that if the author provided `country-field` as the `id` they would get something like `country-field-0.1234567890` as the `id` attribute on the underlying `input`.

Reply:
Interesting point! Thanks for the explanation.
I'll merge it now.

Reply:
Thanks! I hope it helps anyone else who had struggles with that filtering the way I did.

Reply:
I hope too :) You can also write a blog post.